src/pynwb/ecephys.py

```diff
@@ -26,13 +26,29 @@ class ElectrodeGroup(NWBContainer):
             {'name': 'location', 'type': str, 'doc': 'description of location of this electrode group'},
             {'name': 'device', 'type': Device, 'doc': 'the device that was used to record from this electrode group'},
             {'name': 'position', 'type': 'array_data',
-             'doc': 'stereotaxic position of this electrode group (x, y, z)', 'default': None})
+             'doc': 'Compound dataset with stereotaxic position of this electrode group (x, y, z). '
+                    'Each element of the data array must have three elements or the dtype of the '
+                    'array must be `(float, float, float)', 'default': None})
     def __init__(self, **kwargs):
         args_to_set = popargs_to_dict(('description', 'location', 'device', 'position'), kwargs)
         super().__init__(**kwargs)
-        if args_to_set['position'] and len(args_to_set['position']) != 3:
-            raise ValueError('ElectrodeGroup position argument must have three elements: x, y, z, but received: %s'
-                             % str(args_to_set['position']))
+        # position is a compound dataset, i.e., this must be an array with a compound data type of three floats
+        # or each element in the list must at least have three entries
+        if args_to_set['position'] is not None and len(args_to_set['position']) > 0:
+            position_dtype_valid = True
+            if hasattr(args_to_set['position'], 'dtype'):  # If we have a dtype, then check that it is valid
+                if len(args_to_set['position'].dtype) != 3:
```

Comment:
Not sure what the `len` of a `.dtype` here means; is it for structured arrays? (one type per column?)

Reply:
The comments above explain a little bit...

Would this be because of pre-wrapping of the user input by docval before we reach this point?